lib/helpers.bash

```diff
@@ -15,49 +15,49 @@ case "$OSTYPE" in
   'darwin'*) BASH_IT_SED_I_PARAMETERS=(-i "")
 esac
 
-function _command_exists ()
-{
-  _about 'checks for existence of a command'
-  _param '1: command to check'
-  _param '2: (optional) log message to include when command not found'
-  _example '$ _command_exists ls && echo exists'
-  _group 'lib'
-  local msg="${2:-Command '$1' does not exist!}"
-  if type -t "$1" &> /dev/null
-  then
-	  return 0
-  else
-	  _log_warning "$msg"
-	  return 1
-  fi
+function _command_exists() {
+	_about 'checks for existence of a command'
+	_param '1: command to check'
+	_param '2: (optional) log message to include when command not found'
+	_example '$ _command_exists ls && echo exists'
+	_group 'lib'
+	local msg="${2:-Command '$1' does not exist\!}"
```

Comment:
* The escaping of `!` seems inconsistent across the 3 functions - this is the only one that escapes it?

* Do we _need_ to define `msg` in the case whee we don't even show it? (it the success case) - could this be moved inside the `else` ?

Reply:
I would *much* rather remove the default `$msg`...

It's only escaped to appease `shellcheck`, it's not a valid history substitution so _Bash_ doesn't do anything with it.

Reply:
@davidpfarrell, the escaped exclamation mark is fixed now, but if you'll OK it I'd rather remove the default message entirely. Since we're actually using this function now in a lot of places, it's just cluttering up the log (and `bash-it doctor`).